File: DBGuardClient/src/app/interfaces/database-connection-dto.ts
import { DatabaseEngine } from "../enums/database-engines";

export interface DatabaseConnectionDTO {
    id?: number,
    endpoint: string,
    databaseEngine: DatabaseEngine,
    databaseName: string,
    username?: string,
    password?: string,
    createDate: Date,
    lastEdited: Date,
    createdByUserId: string,
    createdByUsername: string
}
export interface SimpleDatabaseConnectionDTO {
    id?: number,
    endpoint: string,
    databaseEngine: DatabaseEngine,
    databaseName: string,
    username?: string,
}
export interface CreateDatabaseConnectionDTO {
    id?: number,
    endpoint: string,
    databaseEngine: DatabaseEngine,
    databaseName: string,
    username?: string,
    password?: string
}

File: DBGuardAPI/Data/DTOs/DatabaseConnectionDTOs/DatabaseConnectionDTO.cs
using DBGuardAPI.Data.Enums;
using Sieve.Attributes;

namespace DBGuardAPI.Data.DTOs.DatabaseConnectionDTOs
{
    public class DatabaseConnectionDTO
    {
        [Sieve(CanSort = true, CanFilter = true)]
        public int Id { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public required string Endpoint { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public DatabaseEngine DatabaseEngine { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public required string DatabaseName { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public string? Username { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public string? Password { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public DateTimeOffset CreateDate { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public DateTimeOffset LastEdited { get; set; }
        public required string CreatedByUserId { get; set; }
        [Sieve(CanSort = true, CanFilter = true)]
        public required string CreatedByUsername { get; set; }
    }
}
